src/util.rs

```diff
@@ -0,0 +1,29 @@
+use encoding_rs::WINDOWS_1252;
+use std::str::from_utf8;
+
+pub trait TakeWord<'a> {
+    fn take_word(&mut self) -> &'a [u8];
+}
+
+impl<'a> TakeWord<'a> for &'a [u8] {
+    fn take_word(&mut self) -> &'a [u8] {
+        if let Some(i) = self.iter().position(|c| c == &b' ') {
+            let word = &self[..i];
+            *self = &self[i + 1..];
+            word
+        } else {
+            let word = &self[..];
+            *self = &self[self.len()..];
+            word
+        }
+    }
+}
+
+pub fn decode_hybrid(data: &[u8]) -> String {
```

Comment:
a note to be made is that, while this may seem hacky, it's the generally accepted best way to handle IRC encoding